### PR TITLE
Replace signatures types with proof systems

### PR DESCRIPTION
**Pull request by willemolding**
**Summary of changes**
Changes introduced in this pull request:
- Removes our mina_crypto::Signature implementation and replaces it with proof_systems::mina_signer::Signature
- Updates deserialization to match
- Removes a test that required the `to_base58` implementation on signature. That no longer makes sense and probably wasn't the standard string format anyway

**Reference issue to close (if applicable)**

Depends on #193 - Merge this first!! 

**Other information and links**

### CODE REVIEW

**Review by hanabi1224 — APPROVED** [review]

File: proof-systems-shim/Cargo.toml [hunk 1]
```diff
@@ -7,5 +7,11 @@ edition = "2021"
...
 
 [dependencies]
-commitment_dlog = {git = "https://github.com/o1-labs/proof-systems", rev = "56517b55920ea7afc811baa541c859dadc8ae50a"}
-mina-hasher = {git = "https://github.com/o1-labs/proof-systems", rev = "56517b55920ea7afc811baa541c859dadc8ae50a"}
+commitment_dlog = {git = "https://github.com/o1-labs/proof-systems", rev = "6af257b793d80f900675acc01b16fd4bf6375268"}
+mina-curves = {git = "https://github.com/o1-labs/proof-systems", rev = "6af257b793d80f900675acc01b16fd4bf6375268"}
+plonk_circuits = {git = "https://github.com/o1-labs/proof-systems", rev = "6af257b793d80f900675acc01b16fd4bf6375268"}
+plonk_protocol_dlog = {git = "https://github.com/o1-labs/proof-systems", rev = "6af257b793d80f900675acc01b16fd4bf6375268"}
+
+mina-hasher = {git = "https://github.com/o1-labs/proof-systems", rev = "1f88a94eddba96b720c9214303fb2b4b0cd97597"}
```
Comment (hanabi1224) [inline proof-systems-shim/Cargo.toml:15]: Possible to use the same rev for all crates?

File: proof-systems-shim/Cargo.toml [hunk 2]
```diff
@@ -7,5 +7,11 @@ edition = "2021"
...
+mina-signer = {git = "https://github.com/o1-labs/proof-systems", rev = "1f88a94eddba96b720c9214303fb2b4b0cd97597"}
```
Comment (hanabi1224) [inline proof-systems-shim/Cargo.toml:16]: duplicate mina-signer

Comment (willemolding) [reply proof-systems-shim/Cargo.toml:16]: is there?

File: proof-systems-shim/Cargo.toml (hunk 1 above)
Comment (willemolding) [reply proof-systems-shim/Cargo.toml:15]: Not directly. The `plonk_*` ones no longer exist in the rev that makes the changes we cant. This definitely means we should no longer use them but we can make that change in another PR

Comment (hanabi1224) [reply proof-systems-shim/Cargo.toml:15]: Actually, I might have removed the usage of stale proof-systems crates in https://github.com/ChainSafe/mina-rs/pull/198

File: proof-systems-shim/Cargo.toml (hunk 2 above)
Comment (hanabi1224) [reply proof-systems-shim/Cargo.toml:16]: nvm : )

File: proof-systems-shim/Cargo.toml [hunk 3]
```diff
@@ -7,6 +7,11 @@ edition = "2021"
 license = "Apache-2.0"
 
 [dependencies]
+commitment_dlog = {git = "https://github.com/o1-labs/proof-systems", rev = "6af257b793d80f900675acc01b16fd4bf6375268"}
```
Comment (willemolding) [inline proof-systems-shim/Cargo.toml:10]:
```suggestion
```

File: proof-systems-shim/Cargo.toml [hunk 4]
```diff
@@ -7,6 +7,11 @@ edition = "2021"
...
+mina-curves = {git = "https://github.com/o1-labs/proof-systems", rev = "6af257b793d80f900675acc01b16fd4bf6375268"}
+plonk_circuits = {git = "https://github.com/o1-labs/proof-systems", rev = "6af257b793d80f900675acc01b16fd4bf6375268"}
+plonk_protocol_dlog = {git = "https://github.com/o1-labs/proof-systems", rev = "6af257b793d80f900675acc01b16fd4bf6375268"}
```
Comment (willemolding) [inline proof-systems-shim/Cargo.toml:13]:
```suggestion
```

File: proof-systems-shim/Cargo.toml (hunk 1 above)
Comment (willemolding) [reply proof-systems-shim/Cargo.toml:12]: Yeah looks like it, that is merged now so all good.

File: proof-systems-shim/Cargo.toml [hunk 5]
```diff
@@ -7,6 +7,8 @@ edition = "2021"
 license = "Apache-2.0"
 
 [dependencies]
+mina-curves = {git = "https://github.com/o1-labs/proof-systems", rev = "6af257b793d80f900675acc01b16fd4bf6375268"}
```
Comment (hanabi1224) [inline proof-systems-shim/Cargo.toml:10]: Ok to use rev 1f88a94eddba96b720c9214303fb2b4b0cd97597 right?

Comment (hanabi1224) [reply proof-systems-shim/Cargo.toml:10]: Or maybe this can be removed, it's an indirect dependency

**Review by creativcoder — APPROVED** [review]

**Pull request merged**